align/compiler/compiler.py

```diff
@@ -199,6 +200,7 @@ def compiler_output(input_ckt, hier_graph_dict, design_name:str, result_dir:path
                     for nm in list(hier_graph_dict.keys()):
                         if nm == lef_name + attr['inst_copy']:
                             if block_name not in hier_graph_dict.keys():
+                                assert False, 'Hope this is dead code. Trying to modify a dictionary while iterating over it!'
```

Comment:
I have seen this code works for some cases. We can keep this assertion for now. Will look into it again after the nightly regression. This does not modify the number of elements in the dict. It adds and removes at same time.

Reply:
It is dangerous even if it keeps the number of elements the same since the location of `block_name` may not be the same as `nm`. We can copy the dict before the iteration if we really need to do this.